src/main/java/de/cityfeedback/feedbackverwaltung/logging/LoggingAspect.java

```diff
@@ -0,0 +1,24 @@
+package de.cityfeedback.feedbackverwaltung.logging;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
```

Comment:
@Aspect dependency direkt aus Springboot würde sich anbieten, da wir ja bereits ein SPringbootprojekt haben. Hier ein Example:

In a Spring Boot application with AOP support, you can utilize aspects to handle logging for domain events in a clean and centralized manner. Here’s a step-by-step guide to implementing logging for domain events using Spring AOP:

1. Add Spring AOP Dependency
Ensure your project includes the necessary Spring AOP dependency. In a Maven project, include:

xml
Code kopieren
<dependency>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-aop</artifactId>
</dependency>
For Gradle:

groovy
Code kopieren
implementation 'org.springframework.boot:spring-boot-starter-aop'
2. Define a Logging Aspect
Create an aspect class to intercept the handling or publishing of domain events. You can use @Aspect for this purpose.

Example: Logging Aspect for Domain Events
java
Code kopieren
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Before;
import org.aspectj.lang.annotation.AfterReturning;
import org.aspectj.lang.annotation.AfterThrowing;
import org.aspectj.lang.annotation.Pointcut;
import org.aspectj.lang.ProceedingJoinPoint;
import org.springframework.stereotype.Component;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Aspect
@Component
public class DomainEventLoggingAspect {

    private static final Logger logger = LoggerFactory.getLogger(DomainEventLoggingAspect.class);

    // Define a pointcut for methods that handle domain events
    @Pointcut("execution(* com.example.yourapp.domain.event..*(..))")
    public void domainEventHandlers() {
        // Pointcut for domain event handlers
    }

    @Before("domainEventHandlers()")
    public void logBefore(ProceedingJoinPoint joinPoint) {
        logger.info("Handling domain event: {}", joinPoint.getSignature());
    }

    @AfterReturning(pointcut = "domainEventHandlers()", returning = "result")
    public void logAfterReturning(ProceedingJoinPoint joinPoint, Object result) {
        logger.info("Successfully handled domain event: {}, Result: {}", joinPoint.getSignature(), result);
    }

    @AfterThrowing(pointcut = "domainEventHandlers()", throwing = "ex")
    public void logAfterThrowing(ProceedingJoinPoint joinPoint, Throwable ex) {
        logger.error("Exception occurred while handling domain event: {}, Error: {}", joinPoint.getSignature(), ex.getMessage());
    }
}
3. Apply the Aspect to Event Handlers
The above example assumes that your domain event handlers follow a specific package structure (e.g., com.example.yourapp.domain.event).

For example, a domain event handler might look like:

java
Code kopieren
@Component
public class OrderCreatedEventHandler {

    public void handle(OrderCreatedEvent event) {
        // Business logic for handling the event
    }
}
4. Customize the Pointcut
You can adjust the @Pointcut definition to match specific annotations or method naming conventions:

Example: Match Methods Annotated with @EventHandler
If your handlers are marked with a custom @EventHandler annotation:

java
Code kopieren
@Pointcut("@annotation(com.example.yourapp.annotations.EventHandler)")
public void eventHandlerMethods() {
}
Example: Match Event Publishing
To log domain events at the point of publishing, define a pointcut targeting your event bus or dispatcher:

java
Code kopieren
@Pointcut("execution(* com.example.yourapp.eventbus.EventBus.publish(..))")
public void eventPublishing() {
}
5. Enable AOP in Spring Boot
AOP is enabled by default in Spring Boot if the AOP starter dependency is included. However, you can explicitly confirm it in your main configuration:

java
Code kopieren
@SpringBootApplication
@EnableAspectJAutoProxy
public class YourApplication {
    public static void main(String[] args) {
        SpringApplication.run(YourApplication.class, args);
    }
}
6. Test Your Setup
Run your application and trigger domain events. You should see logs indicating the handling or publishing of events:

plaintext
Code kopieren
INFO  Handling domain event: OrderCreatedEvent
INFO  Successfully handled domain event: OrderCreatedEvent
ERROR Exception occurred while handling domain event: OrderCreatedEvent, Error: SomeException